fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/ApiRequestImpl.java

```diff
@@ -87,6 +87,7 @@
     protected final ResponseFormatType format;
     protected final Optional<PaginationParameters> paginationParameters;
     protected final long asyncAfter;
+    protected final Optional<String> downloadFilename;
```

Comment:
Don't store as Optional.  Instead store as potentially null and unpack to optional on the way out.

Reply:
why not store as optional?